### PR TITLE
DOC: Improve circular stats doc

### PR DESCRIPTION
**Pull request by dschmitz89**
[skip azp] [skip actions]
#### What does this implement/fix?
Adds explanatory plots to `circvar` and `circstd` functions to make their meaning more clear.

Additionally, a missing space in the References section of `circvar` is fixed: 
Current docs: 
image

### CODE REVIEW

File: scipy/stats/_morestats.py [hunk 1]
```diff
@@ -3922,11 +3942,32 @@ def circstd(samples, high=2*pi, low=0, axis=None, nan_policy='propagate', *,
...
+    >>> import matplotlib.pyplot as plt
+    >>> samples_1 = np.array([0.072, -0.158, 0.077, 0.108, 0.286,
+    ...                       0.133, -0.473, -0.001, -0.348, 0.131])
+    >>> samples_2 = np.array([0.111, -0.879, 0.078, 0.733, 0.421,
+    ...                       0.104, -0.136, -0.867,  0.012,  0.105])
+    >>> circstd_1 = circstd(samples_1)
+    >>> circstd_2 = circstd(samples_2)
+    >>> fig, (left, right) = plt.subplots(ncols=2)
+
+    Plot the samples.
```
Comment (tupui) [inline scipy/stats/_morestats.py:3954]:
```suggestion
    >>> circstd_2 = circstd(samples_2)

    Plot the samples.
```

**Review by tupui — APPROVED** [review]
Thank you @dschmitz89 this really makes it clearer!

Putting this here as I see it. Other improvements for follow ups we could do: add some more text, a general description of all these methods and more `seealso` now and there later.

**Comment by dschmitz89** [comment]
> Thank you @dschmitz89 this really makes it clearer!
> 
> Putting this here as I see it. Other improvements for follow ups we could do: add some more text, a general description of all these methods and more `seealso` now and there later.

Would be happy to help writing this. Where would it go? New section in the general stats tutorial?

**Comment by tupui** [comment]
Great thanks. I am not sure about tutorials because we are having some talks about the possibility to move these to notebooks and another repo etc. @melissawm what would be your suggestion here?

**Pull request merged**